Complete task

4Empty4 opened the pull request:
Add:Bulldozer,Excavator,Machine,Truck classes
Add abstarckt fuction in new classes
Add logs for test

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,13 @@
+public class Excavator extends Machine {
+    @Override
+    public void doWork() {
+        System.out.println("Excavator do work");
```
This message could be more informative. The task description asks for a message indicating that the machine 'started its work'. A message like `Excavator started to work.` would be a better fit.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Excavator.java:
```diff
@@ -0,0 +1,13 @@
+
+    @Override
+    public void stopWork() {
+        System.out.println("Excavator stop work");
```
Similar to the `doWork()` method, this message should clearly state that the machine 'stopped working' as per the task description. For example: `Excavator stopped working.`

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Machine.java:
```diff
@@ -0,0 +1,8 @@
+package core.basesyntax;
+
+public abstract class Machine {
+
```
This empty line violates a coding style requirement. According to checklist item #1, redundant empty lines after a class declaration should be avoided.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,13 @@
+public class Bulldozer extends Machine {
+    @Override
+    public void doWork() {
+        System.out.println("Bulldozer do work");
```
The task requires the message to indicate that the machine has *started* its work. A phrase like "Bulldozer started to work" would align more closely with this requirement.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Bulldozer.java:
```diff
@@ -0,0 +1,13 @@
+
+    @Override
+    public void stopWork() {
+        System.out.println("Bulldozer stop work");
```
Similarly, the requirement is for the message to state that the machine has *stopped* working. Consider changing this to something like "Bulldozer stopped working".

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,13 @@
+public class Truck extends Machine {
+    @Override
+    public void doWork() {
+        System.out.println("Truck do work");
```
This message is functional, but it could be more descriptive. The task asks for a message indicating the machine 'started its work'. A message like `Truck started to work.` would be a better reflection of that requirement.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/Truck.java:
```diff
@@ -0,0 +1,13 @@
+
+    @Override
+    public void stopWork() {
+        System.out.println("Truck stop work");
```
Similarly, this message should clearly state that the machine 'stopped working' as per the task description. For instance: `Truck stopped working.`

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Great job on the overall structure of your project! You've correctly implemented the abstract `Machine` class and the subclasses, and the polymorphism in `MainApp` works as expected. 

However, a few changes are needed to meet all the requirements. The output messages in your `Truck`, `Bulldozer`, and `Excavator` classes should be more descriptive. The task asks for messages indicating that the machine has 'started' and 'stopped' its work. For example, `"Truck do work"` should be changed to something like `"Truck started to work"`. [CHECKLIST ITEM #2]

Additionally, there is a redundant empty line after the class declaration in `Machine.java` that should be removed to adhere to the coding style guidelines. [CHECKLIST ITEM #1]

Once these adjustments are made, your solution will be ready for approval. Keep up the good work!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Excellent work on this submission! I am approving your solution as it perfectly meets all the requirements. You have successfully addressed all the feedback from the previous review, resulting in clean, correct code that demonstrates a solid understanding of abstraction and polymorphism. The descriptive output messages are now spot-on. Great job!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

4Empty4 committed:
Finish task
Add:Bulldozer,Excavator,Machine,Truck classes
Add abstarckt fuction in new classes
Add logs for test

4Empty4 committed:
Small fix mistakes witch AI find
Change
Bulldozer, Excavator, Truck loggs text
